### PR TITLE
[YUNIKORN-2] interface changes for gang

### PR DESCRIPTION
**Pull request by wilfred-s**
Changes for the scheduler interface based on the detailed design
documentation for gang scheduling.
- AddApplication update
- AllocationAsk update
- refactor AllocationRelease Response and Request

### CODE REVIEW

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -437,6 +442,8 @@ message AddApplicationRequest {
   // Execution timeout: How long this application can be in a running state
   // 0 or negative value means never expire.
   int64 executionTimeoutMilliSeconds = 6;
+  // The total amount of resources gang placeholders will request
+  Resource placeholderAsk = 7;
```
Comment (yangwwei) [inline scheduler-interface-spec.md:446]: why we need this?
is this a placeholder for future enhancement, e.g headroom check at queue level?

**Comment by yangwwei** [comment]
Can we use a sub task under YUNIKORN-2 to track this?

File: scheduler-interface-spec.md (hunk 1 above)
Comment (wilfred-s) [reply scheduler-interface-spec.md:446]: It is to make sure that the whole gang fits in the queue, this was already in the design. If the gang does not fit in the queue quota the application will be rejected.
Use case supported: gang total request is 100GB. queue quota is 50 GB. That case we can never allocate the gang, not even in an empty queue and the app will be rejected on submit.

Comment (yangwwei) [reply scheduler-interface-spec.md:446]: Got it, I recall it now.

**Review by yangwwei — APPROVED** [review]
Looks good. Feel free to merge this.
@wilfred-s  can we create a different JIRA for this issue? YUNIKORN-2 is the umbrella we use to track all subtasks.

**Review by kingamarton — COMMENTED** [review]
Looks good to me. I have just just one minor question: shouldn't we use the YUNIKORN-2 branch until the whole feature will be implemented instead of merging the subtasks directly to the master branch?

**Comment by wilfred-s** [comment]
If we push this in a branch we have extra work on merging in the other repos. We need to point to a new commit for the SI after we merge this. While we can now prepare all changes including the go.mod changes in the other repos and not have to think about the SI merge first which would then be followed by dependency fixes in the other repos.

**Pull request closed without merging**